examples/Azure_IoT_Central_ESP32/Azure_IoT_Central_ESP32.ino

```diff
@@ -0,0 +1,600 @@
+#define PST_TIME_ZONE_DAYLIGHT_SAVINGS_DIFF   1
+
+#define GMT_OFFSET_SECS (PST_TIME_ZONE * 3600)
+#define GMT_OFFSET_SECS_DST ((PST_TIME_ZONE + PST_TIME_ZONE_DAYLIGHT_SAVINGS_DIFF) * 3600)
+
+#define UNIX_TIME_NOV_13_2017 1510592825
+#define UNIX_EPOCH_START_YEAR 1900
+
+/* --- Function Returns --- */
+#define RESULT_OK       0
+#define RESULT_ERROR    __LINE__
+
+/* --- Handling iot_config.h Settings --- */
+static const char* wifi_ssid = IOT_CONFIG_WIFI_SSID;
+static const char* wifi_password = IOT_CONFIG_WIFI_PASSWORD;
+
+// TODO: remove this after updating library with Azure SDK for C version 1.3.0-beta.2 or later.
+#define AZURE_SDK_CLIENT_USER_AGENT_WORKAROUND "DeviceClientType=" AZURE_SDK_CLIENT_USER_AGENT
+
+/* --- Function Declarations --- */
+static void sync_device_clock_with_ntp_server();
+static void connect_to_wifi();
+static esp_err_t esp_mqtt_event_handler(esp_mqtt_event_handle_t event);
+
+// This is a logging function used by Azure IoT client.
+static void logging_function(log_level_t log_level, char const* const format, ...);
+
+/* --- Sample variables --- */
+static azure_iot_config_t azure_iot_config;
+static azure_iot_t azure_iot;
+static esp_mqtt_client_handle_t mqtt_client;
+
+static char mqtt_broker_uri[128];
+
+#define AZ_IOT_DATA_BUFFER_SIZE 1000 
+static uint8_t az_iot_data_buffer[AZ_IOT_DATA_BUFFER_SIZE];
+
+#define MQTT_PROTOCOL_PREFIX "mqtts://"
+
+static uint32_t properties_request_id = 0;
+static bool send_device_info = true;
+
+
+/* --- MQTT Interface Functions --- */
+/*
+ * These functions are used by Azure IoT to interact with whatever MQTT client used by the sample
+ * (in this case, Espressif's ESP MQTT). Please see the documentation in AzureIoT.h for more details.
+ */
+
+/*
+ * See the documentation of `mqtt_client_init_function_t` in AzureIoT.h for details.
+ */
+static int mqtt_client_init_function(mqtt_client_config_t* mqtt_client_config, mqtt_client_handle_t *mqtt_client_handle)
+{
+  int result;
+  esp_mqtt_client_config_t mqtt_config;
+  memset(&mqtt_config, 0, sizeof(mqtt_config));  
+
+  az_span mqtt_broker_uri_span = AZ_SPAN_FROM_BUFFER(mqtt_broker_uri);
+  mqtt_broker_uri_span = az_span_copy(mqtt_broker_uri_span, AZ_SPAN_FROM_STR(MQTT_PROTOCOL_PREFIX));
+  mqtt_broker_uri_span = az_span_copy(mqtt_broker_uri_span, mqtt_client_config->address);
+  az_span_copy_u8(mqtt_broker_uri_span, null_terminator);
+
+  mqtt_config.uri = mqtt_broker_uri;
+  mqtt_config.port = mqtt_client_config->port;
+  mqtt_config.client_id = (const char*)az_span_ptr(mqtt_client_config->client_id);
+  mqtt_config.username = (const char*)az_span_ptr(mqtt_client_config->username);
+  mqtt_config.password = (const char*)az_span_ptr(mqtt_client_config->password);
+  mqtt_config.keepalive = 30;
+  mqtt_config.disable_clean_session = 0;
+  mqtt_config.disable_auto_reconnect = false;
+  mqtt_config.event_handle = esp_mqtt_event_handler;
+  mqtt_config.user_context = NULL;
+  mqtt_config.buffer_size = 1024;
```

Comment:
how is this number being determined? Is this malloc'd underneath?

Reply:
It's malloc'ed underneath, but that's defined by ESP MQTT documentation. Just following their guidelines.